sandbox/src/main/java/org/robolectric/interceptors/AndroidInterceptors.java

```diff
@@ -492,8 +492,13 @@ public Object call(Class<?> theClass, Object value, Object[] params) {
     @Override
     public MethodHandle getMethodHandle(String methodName, MethodType type)
         throws NoSuchMethodException, IllegalAccessException {
-      return lookup.findStatic(
-          getClass(), "getCurrentSpi", methodType(CipherSpi.class, void.class));
+      switch (methodName) {
+        case "getCurrentSpi":
+          return lookup.findStatic(
+              Cipher.class, methodName, methodType(CipherSpi.class, void.class));
+        default:
+          throw new IllegalStateException();
```

Comment:
Maybe `UnsupportedOperationException` is better. IIRC, this file has similar usage.

Reply:
OK, other interceptors in the file throw this exception though

Reply:
I didn't change this as per my comment. Let me know if I should